File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.js
function solution(arr) {
  const answer = [];
  let prev = arr[0];
  answer.push(prev);

  for (let i = 1; i < arr.length; i++) {
    if (arr[i] !== prev) {
      answer.push(arr[i]);
      prev = arr[i];
    }
  }

  return answer;
}

File: 프로그래머스/1/12906. 같은 숫자는 싫어/같은 숫자는 싫어.java
import java.util.ArrayList;

public class Solution {
    public int[] solution(int[] arr) {
        ArrayList<Integer> list = new ArrayList<>();
        int prev = arr[0];
        list.add(prev);

        for (int i = 1; i < arr.length; i++) {
            if (arr[i] != prev) {
                list.add(arr[i]);
                prev = arr[i];
            }
        }

        int[] answer = new int[list.size()];
        for (int i = 0; i < list.size(); i++) {
            answer[i] = list.get(i);
        }

        return answer;
    }
}
